pkg/controllers/util/work.go

```diff
@@ -1,18 +1,35 @@
-package helper
+package util
 
 import (
 	"context"
+	"fmt"
+	"hash/fnv"
+	"strings"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/rand"
 	"k8s.io/klog/v2"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
 	workv1alpha1 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha1"
 )
 
+// GenerateWorkName will generate work name by its name and the hash of its namespace, kind and name.
+func GenerateWorkName(kind, name, namespace string) string {
```

Comment:
I think it's about named functions, and it feels good with other named functions.

Reply:
sorry, I don't get your point.Could you clarify it more precisely?

Reply:
This function maybe can still in `names.go` file, the functions in this file are all about nameing and are easy to understand and query.

Reply:
since it's only used in pkg/controllers.IMO, if function is used in multiple module,such as controller and scheduler,then pkg/util may be the right place.